dali/operators/reader/loader/video/frames_decoder_gpu.cc

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/reader/loader/video/frames_decoder_gpu.h"
+
+#include <cuda.h>
+#include <unistd.h>
+
+#include <string>
+#include <memory>
+
+#include "dali/core/error_handling.h"
+#include "dali/core/cuda_utils.h"
+#include "dali/pipeline/data/backend.h"
+#include "dali/pipeline/data/tensor.h"
+#include "dali/operators/reader/loader/video/nvdecode/ColorSpace.h"
+
+namespace dali {
+namespace detail {
+int process_video_sequence(void *user_data, CUVIDEOFORMAT *video_format) {
+  return video_format->min_num_decode_surfaces;
+}
+
+int process_picture_decode(void *user_data, CUVIDPICPARAMS *picture_params) {
+  FramesDecoderGpu *frames_decoder = static_cast<FramesDecoderGpu*>(user_data);
+  if (frames_decoder->flush_) {
+    frames_decoder->flush_ = false;
+    frames_decoder->decode_success_ = false;
+    return 0;
```

Comment:
Can you use the right enum here?

Reply:
The only piece docs on this I could find says:
```
/* Return value from HandlePictureDecode() are interpreted as:
 *   0: fail
 *   >=1: suceeded
 */
```